types/coin.go

```diff
@@ -592,7 +592,7 @@ var (
 	// Denominations can be 3 ~ 16 characters long.
 	reDnmString = `[a-z][a-z0-9]{2,15}`
 	reAmt       = `[[:digit:]]+`
-	reDecAmt    = `[[:digit:]]*\.[[:digit:]]+`
+	reDecAmt    = `[[:digit:]]*\.?[[:digit:]]+`
```

Comment:
This is about parsing a string representation of a decimal amount, therefore this has to be a decimal number.
When it comes to number parsing, the stricter the syntax the better.